### ISSUE TITLE
Blob I/O

### ISSUE THREAD

**Issue by smhmw** [mattn/go-sqlite3#239]
Would it be possible to implement the blob i/o API just as the backup API ?

[mxk/go-sqlite](https://github.com/mxk/go-sqlite/tree/master/sqlite3) has it but is a little out of date.
[gwenn/gosqlite](https://github.com/gwenn/gosqlite) has it too.

**Comment by mattn** [comment]
database/sql doesn't provide Blob APIs. they provide Blob APIs for native driver (not database/sql).

**Comment by smhmw** [comment]
Yes, but we can provide it aside from database/sql, like backup (backup.go) ? I may try to make it work locally, if it works i'll submit the patch - if that's interesting enough.

**Comment by mattn** [comment]
No way to pass the Blob type into database.sql.

**Comment by gwenn** [comment]
To make it work, you need access from `sql.DB` to the underlying/wrapped `sqlite3.SQLiteConn`...

**Comment by smhmw** [comment]
I don't want to access it from `sql.DB`, rather add an API to the lib in a similar way to `sqlite3.SQLiteBackup`, such as `sqlite3.SQLiteBlob`; and corresponding method in `sqlite3.SQLiteConn` such as `Blob(db, table, column, rowid)` (similar to `SQLiteConn.Backup(...)`)

I'll give it a try ...

**Comment by anacrolix** [comment]
:+1:

**Comment by anacrolix** [comment]
Any progress on this @smhmw ?

**Comment by GJRTimmer** [comment]
@smhmw Any update ?

**Comment by ncruces** [comment]
I think I came up with a nice API for this in my SQLite driver, which could be implemented for this driver as well:
https://github.com/ncruces/go-sqlite3/blob/main/ext/blob/blob_test.go

This would require wrapping the [pointer passing](https://www.sqlite.org/bindptr.html) APIs, on top of blob IO.
